mmv1/third_party/tgc_next/test/assert_test_files.go

```diff
@@ -34,59 +34,77 @@ var (
 )
 
 func BidirectionalConversion(t *testing.T, ignoredFields []string, ignoredAssetFields []string) {
-	retries := 0
-	flakyAction := func(ctx context.Context) error {
-		log.Printf("Starting the retry %d", retries)
-		resourceTestData, primaryResource, err := prepareTestData(t.Name(), retries)
-		retries++
-		if err != nil {
-			return fmt.Errorf("error preparing the input data: %v", err)
-		}
+	testName := t.Name()
+	stepNumbers, err := getStepNumbers(testName)
+	if err != nil {
+		t.Fatalf("error preparing the input data: %v", err)
+	}
 
-		if resourceTestData == nil {
-			return retry.RetryableError(fmt.Errorf("fail: test data is unavailable"))
-		}
+	if len(stepNumbers) == 0 {
+		t.Skipf("test steps are unavailable")
+	}
 
-		// Create a temporary directory for running terraform.
-		tfDir, err := os.MkdirTemp(tmpDir, "terraform")
-		if err != nil {
-			return err
-		}
-		defer os.RemoveAll(tfDir)
+	// Create a temporary directory for running terraform.
+	tfDir, err := os.MkdirTemp(tmpDir, "terraform")
+	if err != nil {
+		t.Fatalf("error creating a temporary directory for running terraform: %v", err)
+	}
+	defer os.RemoveAll(tfDir)
 
-		logger := zaptest.NewLogger(t)
+	logger := zaptest.NewLogger(t)
 
-		// If the primary resource is specified, only test the primary resource.
-		// Otherwise, test all of the resources in the test.
-		if primaryResource != "" {
-			t.Logf("Test for the primary resource %s begins.", primaryResource)
-			err = testSingleResource(t, t.Name(), resourceTestData[primaryResource], tfDir, ignoredFields, ignoredAssetFields, logger, true)
-			if err != nil {
-				return err
-			}
-		} else {
-			for _, testData := range resourceTestData {
-				err = testSingleResource(t, t.Name(), testData, tfDir, ignoredFields, ignoredAssetFields, logger, false)
+	for _, stepN := range stepNumbers {
+		subtestName := fmt.Sprintf("step%d", stepN)
+		t.Run(subtestName, func(t *testing.T) {
```

Comment:
Out of curiousity, why does this function need to be defined anonymously?

Reply:
This function refers some variables from the outer function. 

I will double check if it works to make it a separate function.